core/base/src/TListOfTypes.cxx

```diff
@@ -62,29 +62,20 @@ static bool NameExistsElsewhere(const char* name){
    if (lastPos != nullptr) {
       // We have a scope
       const auto enName = lastPos + 1;
-      const auto scopeNameSize = ((Long64_t)lastPos - (Long64_t)name) / sizeof(decltype(*lastPos)) - 1;
-#ifdef R__WIN32
-      char *scopeName = new char[scopeNameSize + 1];
-#else
-      char scopeName[scopeNameSize + 1]; // on the stack, +1 for the terminating character '\0'
-#endif
-      strncpy(scopeName, name, scopeNameSize);
-      scopeName[scopeNameSize] = '\0';
+      const auto scopeNameSize = static_cast<unsigned int>(((Long64_t)lastPos - (Long64_t)name) / sizeof(decltype(*lastPos)) - 1);
```

Comment:
What was the previous version?  Wouldn't the following be the 'best'?:
```suggestion
      const auto scopeNameSize = (((size_t)lastPos - (size_t)name) / sizeof(decltype(*lastPos)) - 1);
```
or even
```suggestion
      const size_t scopeNameSize = (lastPos - name) / sizeof(decltype(*lastPos)) - 1;
```